docs/user/attestations/internals.md

```diff
@@ -0,0 +1,335 @@
+*Trusted Publisher* identities. In practice, this means that the identity
+expected to sign a distribution's attestation is expected to match the
+Trusted Publisher that published the package.
+
+For example, for a GitHub-based Trusted Publisher, the identity might be
+`https://github.com/pypa/sampleproject/blob/main/.github/workflows/release.yml`,
+i.e. `pypa/sampleproject` on GitHub, publishing from a workflow defined
+in `release.yml`.
+
+### Future identities
+
+In the future, PyPI may allow signing identities other than the project's
+Trusted Publishers. Some potential future signing identities include:
+
+* E-mail addresses, checked against the project's owners' profiles
+  and/or against the uploaded distribution's own metadata.
+* Third-party identities (such as GitHub or GitLab usernames), checked against
+  the uploaded distribution's own metadata.
+
+## Attestation types
+
+The "scope" of the signing identity varies with the different attestation
+types that can be uploaded to PyPI.
+
+### PyPI Publish Attestation
+
+A [PyPI Publish Attestation](/attestations/publish/v1/) is intended to
+attest to the Trusted Publisher itself. Therefore, the identity used
+is exactly the identity of the Trusted Publisher itself.
+
+For example, using the GitHub-based Trusted Publisher above, the
+expected signing identity will be **exactly**
+`https://github.com/pypa/sampleproject/blob/main/.github/workflows/release.yml`.
+
+### SLSA Provenance
+
+[SLSA Provenance](https://slsa.dev/spec/v1.0/provenance) is intended to more
+generally trace a software artifact back to its source.
+
+Because of this, the identity used to verify a SLSA Provenance attestation
+is slightly looser than for a PyPI Publish Attestation: any
+identity under `https://github.com/pypa/sampleproject` is accepted, not just
+ones corresponding to the `release.yml` workflow.
+
+This is intended to reflect common CI/CD pipeline patterns: `release.yml`
+is not itself necessarily responsible for producing the distribution that
+gets published, and so SLSA Provenance can't be assumed to be tightly bound to
+it.
+
+Consequently, downstream consumers/verifiers of SLSA Provenance attestations
+may wish to further evaluate the attestation payload and signing identity
+on a local policy basis.
+
+### Future attestations
+
+Per [future identities](#future-identities), PyPI may allow additional
+attestation types in the future. These future types may depend on future
+identities. Some potential future attestation types:
+
+* Third-party review attestations, e.g. manual review or automatic scanning
+  from security tooling.
+* "Release" attestations from the index itself, attesting that PyPI has
+  actually made a particular distribution available for public download.
+
+## Attestation object internals
+
+This section is intended as a high-level walkthrough of a PEP 740 attestation
+object.
+
+First: here is our contrived attestation object:
+
+```json
+{
+    "version": 1,
```

Comment:
Should this be `media_type`?

Reply:
This is the attestation object layout, which gets (losslessly) produced from a Sigstore bundle: https://peps.python.org/pep-0740/#attestation-objects

(We did this to ensure that the format is, in principle, reusable outside of a Sigstore bundle context.)

You can see an example conversion routine here: https://github.com/trailofbits/pypi-attestation-models/blob/b7ede39f47c5a170a816491065941b5b8b16c8ea/src/pypi_attestation_models/_impl.py#L175-L196

Reply:
Makes sense, thanks!